core/unit_test/TestGraph.hpp

```diff
@@ -398,6 +400,103 @@ TEST_F(TEST_CATEGORY_FIXTURE(graph), empty_graph) {
   ex.fence();
 }
 
+template <typename ViewType, size_t size,
+          typename = std::enable_if_t<(ViewType::rank() == 0)>>
+struct ForceGlobalLaunchFunctor {
+ public:
+  ViewType data;
+
+  ForceGlobalLaunchFunctor(ViewType data_) : data(std::move(data_)) {}
+
+  template <typename T>
+  KOKKOS_FUNCTION void operator()(const T) const {
+    ++data();
+  }
+
+ private:
+  Kokkos::Array<typename ViewType::value_type, size> unused{};
```

Comment:
```suggestion
  static constexpr size_t count =
  #if defined(KOKKOS_ENABLE_CUDA)
  Kokkos::Impl::CudaTraits::ConstantMemoryUsage
  #elif defined(KOKKOS_ENABLE_HIP)
  Kokkos::Impl::HIPTraits::ConstantMemoryUsage
  #endif
  + 1;
  #endif
  std::byte unused[count] = {};
```

Reply:
Indeed, your suggestion makes it much simpler to reason about the code. Also, it is not the functor itself that decides what size is needed for enforcing global launches, and that makes even more sense.